SwiftCharts/Layers/ChartGroupedBarsLayer.swift

```diff
@@ -88,8 +88,8 @@ open class ChartGroupedBarsLayer<T: ChartBarModel, U: ChartPointViewBar>: ChartC
         let totalWidth = CGFloat(group.bars.count) * barWidth + ((barSpacing ?? 0) * (maxBarCountInGroup - 1))
         let groupCenter = screenLocCalculator(group.constant.scalar)
         let origin = groupCenter - totalWidth / 2
-        let barWidthAddition = (barWidth + (barSpacing ?? 0)) + barWidth / 2
-        return origin + CGFloat(index) * barWidthAddition
+        let calculatedWidthWithSpacing = CGFloat(index) * (barWidth + (barSpacing ?? 0))
```

Comment:
I'd call this something like internalBarOriginScreenLoc. It refers to the position of the bar in the screen, not width.

Reply:
Do change on **calculateConstantScreenLoc** function

    func calculateConstantScreenLoc(_ screenLocCalculator: (Double) -> CGFloat, index: Int, group: ChartPointsBarGroup<T>, barWidth: CGFloat, maxBarCountInGroup: CGFloat) -> CGFloat {
        let totalWidth = CGFloat(group.bars.count) * barWidth + ((barSpacing ?? 0) * (maxBarCountInGroup - 1))
        let groupCenter = screenLocCalculator(group.constant.scalar)
        let origin = groupCenter - totalWidth / 2
        let calculatedWidthWithSpacing = CGFloat(index) * (barWidth + (barSpacing ?? 0))
        return origin + calculatedWidthWithSpacing + barWidth / 2
    }